WordPress/src/jetpack/assets/bot_widget.js

```diff
@@ -1,32 +1,67 @@
-;
-window.DocsBotAI = window.DocsBotAI || {}, DocsBotAI.init = function(c) {
-    return new Promise(function(e, o) {
-        var t = document.createElement("script");
-        t.type = "text/javascript", t.async = !0, t.src = "https://widget.docsbot.ai/chat.js";
-        var n = document.getElementsByTagName("script")[0];
-        n.parentNode.insertBefore(t, n), t.addEventListener("load", function() {
+window.DocsBotAI = window.DocsBotAI || {};
+
+DocsBotAI.init = function (c) {
+    return new Promise(function (resolve, reject) {
+        const chatScript = document.createElement("script");
+        chatScript.type = "text/javascript";
+        chatScript.async = true;
+        chatScript.src = "https://widget.docsbot.ai/chat.js";
+
+        const firstScript = document.getElementsByTagName("script")[0];
+        firstScript.parentNode.insertBefore(chatScript, firstScript);
+
+        chatScript.addEventListener("load", function () {
             window.DocsBotAI.mount({
                 id: c.id,
                 supportCallback: c.supportCallback,
                 identify: c.identify,
                 options: c.options,
                 signature: c.signature
             });
-            var t;
-            t = function(n) {
-                return new Promise(function(e) {
-                    if (document.querySelector(n)) return e(document.querySelector(n));
-                    var o = new MutationObserver(function(t) {
-                        document.querySelector(n) && (e(document.querySelector(n)), o.disconnect())
-                    });
-                    o.observe(document.body, {
-                        childList: !0,
-                        subtree: !0
-                    })
-                })
-            }, t && t("#docsbotai-root").then(e).catch(o)
-        }), t.addEventListener("error", function(t) {
-            o(t.message)
-        })
-    })
+
+            const waitForElementAndShadowRoot = function (selector) {
+                return new Promise(function (resolve) {
+                    const elementCheck = function (mutationsList, observer) {
+                        const shadowRoot = document.querySelector(selector)?.shadowRoot;
+                        if (shadowRoot) {
+                            // reset chat history
+                            localStorage.removeItem("docsbot_chat_history");
+
+                            // Open DocsBotAI after shadowRoot is loaded
+                            window.DocsBotAI.open();
+
+                            // Observe the shadowRoot for changes (including header appearance)
+                            const shadowRootObserver = new MutationObserver(function (mutationsList, observer) {
+                                const header = shadowRoot.querySelector("div > div > div > div.docsbot-chat-header");
+                                const closeButton = shadowRoot.querySelector("div > div > div > a");
```

Comment:
At least from css, I found that I can query these elements by class names, such as `docsbot-chat-header,` and `mobile-close-button`. I haven't rechecked if it works with `querySelector`. For now, this approach works. If class names work as well, it could be a safer option.

Reply:
Note: I also tried to hide elements using CSS. It works the same way. We still need to load shadowRoot though before appending a new CSS, so I don't have strong opinion if any of these solutions is better or worse than the other: https://github.com/wordpress-mobile/WordPress-iOS/pull/21314/commits/f866e31476da41c5dce4e3c717b38c11ed3fc3aa

Reply:
Awesome find, switched to using CSS. 🎉